python/ray/data/read_api.py

```diff
@@ -2282,21 +2283,28 @@ def from_tf(
 @PublicAPI
 def from_torch(
     dataset: "torch.utils.data.Dataset",
+    parallelism: int = -1,
+    shuffle: bool = False,
+    batch_size: int = 32,
 ) -> MaterializedDataset:
     """Create a :class:`~ray.data.Dataset` from a
     `Torch Dataset <https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset/>`_.
 
-    This function is inefficient. Use it to read small datasets or prototype.
-
     .. warning::
         If your dataset is large, this function may execute slowly or raise an
         out-of-memory error. To avoid issues, read the underyling data with a function
         like :meth:`~ray.data.read_images`.
 
     .. note::
-        This function isn't parallelized. It loads the entire dataset into the head
+        This function uses a streaming implementation. The dataset does not need to fit completely into memory.
+
+    .. note::
+        For iterable-style datasets, this function is not parallelized. It loads the entire dataset into the head
         node's memory before moving the data to the distributed object store.
 
+    .. note::
+        Parallel read dispatches read tasks to worker nodes. The underlying files of the dataset must be accesible from every node in the cluster.
```

Comment:
I think it's pretty common that the data is only accessible on the local, right? E.g. when setting `download=True`. 
I'm not sure if torch data has an API to tell if the data is local only. If not, we should add a parameter that allows users to force running the read tasks on local node.